### PR TITLE
:star:  Expand data collected in aws.vpcs {routeTables {associations}}

### PR DESCRIPTION
**Pull request by mm-weber**
adds missing info, so we can connect `subnets` to `routeTables`

image

### CODE REVIEW

**Comment by vjeffrey** [comment]
is this ready for review @mm-weber ?

File: providers/aws/resources/aws.lr [hunk 1]
```diff
@@ -88,6 +90,22 @@ private aws.vpc.routetable @defaults("id routes.length") {
   tags map[string]string
 }
 
+// Amazon Virtual Private Cloud (VPC) route table associations
+private aws.vpc.routetable.associations @defaults("routeTableAssociationId gatewayId subnetId") {
+  // Unique ID of the association ID
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Unique ID of the association
```

File: providers/aws/resources/aws.lr [hunk 2]
```diff
@@ -88,6 +90,22 @@ private aws.vpc.routetable @defaults("id routes.length") {
...
+  routeTableAssociationId
+  // the association state
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Association state
```

File: providers/aws/resources/aws.lr [hunk 3]
```diff
@@ -88,6 +90,22 @@ private aws.vpc.routetable @defaults("id routes.length") {
...
+  associationsState []dict
+  // Unique ID of the associated gateway
+  gatewayId string
+  // Boolean if association is the main association
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether this is the main association
```

File: providers/aws/resources/aws.lr [hunk 4]
```diff
@@ -88,6 +90,22 @@ private aws.vpc.routetable @defaults("id routes.length") {
...
+  main string
+  // Unique ID of the route table
+  routeTableId
+  // The subnet of the route table association
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Subnet of the route table association
```

**Review by misterpantz — COMMENTED** [review]
@mm-weber this looks good! Just a few suggestions in the docs for consistency. 😃

File: providers/aws/resources/aws_vpc.go [hunk 5]
```diff
@@ -546,11 +546,16 @@ func (a *mqlAwsVpc) routeTables() ([]interface{}, error) {
 			if err != nil {
 				return nil, err
 			}
+			dictAssociations, err := convert.JsonToDictSlice(routeTable.Associations)
+			if err != nil {
+				return nil, err
+			}
 			mqlRouteTable, err := CreateResource(a.MqlRuntime, "aws.vpc.routetable",
 				map[string]*llx.RawData{
-					"id":     llx.StringDataPtr(routeTable.RouteTableId),
```
Comment (vjeffrey) [inline providers/aws/resources/aws_vpc.go]: @mm-weber is some of your code missing? I don't see the associations resource here?

Comment (mm-weber) [reply providers/aws/resources/aws_vpc.go]: Thanks for the review and the hint @vjeffrey , will get back on this on the weekend.

File: providers/aws/resources/aws.lr (hunk 1 above)
Comment (mm-weber) [reply providers/aws/resources/aws.lr]: Thanks for the input, resolved!

File: providers/aws/resources/aws.lr (hunk 3 above)
Comment (mm-weber) [reply providers/aws/resources/aws.lr]: Thank you

**Comment by mm-weber** [comment]
@vjeffrey still having issues completing this properly:

```
--> [aws] generate CLI json
# go.mondoo.com/cnquery/v11/providers/aws/resources
resources/aws.lr.go:10402:12: c.associations undefined (type *mqlAwsVpcRoutetable has no field or method associations, but does have Associations)
resources/aws.lr.go:10495:12: c.subnet undefined (type *mqlAwsVpcRoutetableAssociations has no field or method subnet, but does have Subnet)
make: *** [Makefile:338: providers/build/aws] Error 1

```

**Comment by vjeffrey** [comment]
@mm-weber  ya, it won't compile to the resource code is actually added, I'm working on that this evening

**Comment by mm-weber** [comment]
Thanks @vjeffrey for all your help and work on this! Will pin it as a reference for myself.

Managed to fix the last few missing details:

image


Looks ready to go now

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 6]
```diff
@@ -2938,6 +2941,31 @@ resources:
...
+  aws.vpc.routetable.association:
+    fields:
+      associationsState: {}
+      gatewayId: {}
+      main: {}
+      routeTableAssociationId: {}
+      routeTableId: {}
+      subnet: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 7]
```diff
@@ -2938,6 +2941,31 @@ resources:
...
+      main: {}
+      routeTableAssociationId: {}
+      routeTableId: {}
+      subnet: {}
+    is_private: true
+    min_mondoo_version: 9.0.0
+    platform:
+      name:
+      - aws
+  aws.vpc.routetable.associations:
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]: Is this meant to be here in additional to the singular one above?

**Pull request merged**